src/Storages/MergeTree/DataPartsExchange.cpp

```diff
@@ -462,6 +604,103 @@ MergeTreeData::MutableDataPartPtr Fetcher::downloadPartToDisk(
     return new_data_part;
 }
 
+MergeTreeData::MutableDataPartPtr Fetcher::downloadPartToS3(
+    const String & part_name,
+    const String & replica_path,
+    bool to_detached,
+    const String & tmp_prefix_,
+    const Disks & disks_s3,
+    PooledReadWriteBufferFromHTTP & in
+    )
+{
+    if (disks_s3.empty())
+        throw Exception("No S3 disks anymore", ErrorCodes::LOGICAL_ERROR);
+
+    String part_id;
+    readStringBinary(part_id, in);
+
+    DiskPtr disk = disks_s3[0];
+
+    for (const auto & disk_s3 : disks_s3)
+    {
+        if (disk_s3->checkUniqueId(part_id))
+        {
+            disk = disk_s3;
+            break;
+        }
+    }
+
+    if (!disk)
```

Comment:
622: `DiskPtr disk = disks_s3[0];`

How is it possible?

Reply:
Unuseful check for now.